linker/shared/src/test/scala/org/scalajs/linker/OptimizerTest.scala

```diff
@@ -562,6 +562,50 @@ class OptimizerTest {
     }
   }
 
+  @Test
+  def removeUnreachableLinkTimeIfBranch(): AsyncResult = await {
+    val methodName = m("method", Nil, I)
+    val classDefs = Seq(
+      classDef("Foo", kind = ClassKind.Class, superClass = Some(ObjectClass),
+        methods = List(
+          trivialCtor("Foo"),
+          // def method(): Int = 0
+          MethodDef(EMF, methodName, NON, Nil, IntType, Some(int(0)))(EOH, UNV))
+      ),
+      mainTestClassDef({
+        LinkTimeIf(
+          LinkTimeCondition.Binary(
+            BinaryOp.Boolean_==,
+            LinkTimeValue.Property(
+                "scala.scalajs.LinkingInfo.productionMode", BooleanType),
+            LinkTimeValue.BooleanConst(true)),
+          consoleLog(str("prod")),
+          consoleLog(Apply(EAF, New("Foo", NoArgConstructorName, Nil),
+              methodName, Nil)(IntType))
+        )(NoType)
+      })
+    )
+
+    for {
+      // TODO: it fails at IR cheker
+      // (1:1:New): Cannot find class Foo
+      moduleSet <- linkToModuleSet(
+        classDefs, MainTestModuleInitializers,
+        config = StandardConfig().withSemantics((_.withProductionMode(true)))
+      )
+    } yield {
+      findClass(moduleSet, MainTestClassName.nameString).get
+        .methods.find(_.name.name == MainMethodName).get
+        .body.get match {
+          case t if t == consoleLog(str("prod")) =>
+            // ok
+
+          case t =>
+            fail(s"Unexpected body: $t")
+        }
+    }
+  }
```

Comment:
```scala
class Foo {
  def method(): Int = 0
}

class Test {
  LinkingInfo.linkTimeIf(LInkingInfo.productionMode /* true */) {
    console.log("prod")
  } {
    console.log((new Foo).method())
  }
}
```

This fails at IRChecker with `(1:1:New): Cannot find class Foo` because

- The `Analyzer` evaluates `LinkTimeIf`, but it doesn't mark `Foo` as reachable because the else branch will be eliminated later.
- `BaseLinker.assemble` removes `Foo` because it's unreachable.
- `IRChecker` checks the else branch of `LinkTimeIf`, which contains `new Foo`. However, looking up `Foo` fails because it has been removed by the assembler.

A potential workaround for this issue would be to modify `IRChecker` to evaluate `LinkTimeIf` and check only the viable branch. It could then ignore the dead branch since it will be removed later in the optimizer or backend.

What do you think about this approach?

Would you like me to explain or elaborate on any part of this revision?